### PR TITLE
Log in subtests similar to stdout

### PR DESCRIPTION
**Pull request by rhoban13**
Logging inside subtests is not displayed unless the parent test fails.  This PR proposes capturing logging for each subtest similar to how stdout/stderr is currently captured and displaying with similar granularity.

fixes https://github.com/pytest-dev/pytest-subtests/issues/87

### CODE REVIEW

File: pytest_subtests.py [hunk 1]
```diff
@@ -155,14 +156,24 @@ def _capturing_output(self):
                 fixture.close()
                 captured.out = out
                 captured.err = err
+    
+    @contextmanager
+    def _capturing_logs(self):
+        logging_plugin = self.request.config.pluginmanager.getplugin("logging-plugin")
```
Comment (nicoddemus) [inline pytest_subtests.py:162]: We should probably take in account that the logging plugin might be disabled via `-p no:logging`, so this call would return `None` and crash two lines below.

Could you account for that and include an additional test? Thanks!

File: pytest_subtests.py [hunk 2]
```diff
@@ -155,14 +156,24 @@ def _capturing_output(self):
...
+            yield captured_logs
 
     @contextmanager
     def test(self, msg=None, **kwargs):
         start = time.time()
         precise_start = time.perf_counter()
         exc_info = None
 
-        with self._capturing_output() as captured:
+        with self._capturing_output() as captured, self._capturing_logs() as capturedlogs:
```
Comment (nicoddemus) [inline pytest_subtests.py:176]: Minor change for better readability:

```suggestion
        with self._capturing_output() as captured_output, self._capturing_logs() as captured_logs:
```

File: tests/test_subtests.py [hunk 3]
```diff
@@ -365,6 +365,65 @@ def test(subtests, {fixture}):
...
+                        with subtests.test("sub1"):
+                            print("sub1 stdout")
+                            logging.info("sub1 logging")
+
+                        with subtests.test("sub2"):
+                            print("sub2 stdout")
+                            logging.info("sub2 logging")
+                            assert False
+                """
+        )
```
Comment (nicoddemus) [inline tests/test_subtests.py:386]: Fixing indentation:

```suggestion
        testdir.makepyfile(
            """
            import logging

            def test_foo(subtests):
                logging.info("before")

                with subtests.test("sub1"):
                    print("sub1 stdout")
                    logging.info("sub1 logging")

                with subtests.test("sub2"):
                    print("sub2 stdout")
                    logging.info("sub2 logging")
                    assert False
            """
        )
```

**Review by nicoddemus — CHANGES_REQUESTED** [review]
Thanks a lot @rhoban13, this looks excellent!

Besides my comments, could you please update the CHANGELOG? Thanks!

I will be happy to make a new release once this is merged. 👍

File: pytest_subtests.py (hunk 1 above)
Comment (rhoban13) [reply pytest_subtests.py]: Good catch - I updated with one approach, let me know what you think.

File: src/pytest_subtests.py [hunk 4]
```diff
@@ -176,14 +177,27 @@ def _capturing_output(self):
                 fixture.close()
                 captured.out = out
                 captured.err = err
+    
+    @contextmanager
```
Comment (rhoban13) [inline src/pytest_subtests.py:181]: Noticing the comment on line 154 above "# capsys or capfd are active, subtest should not capture".  Do I need to do anything similar if the `caplog` fixture is active?

Comment (nicoddemus) [reply src/pytest_subtests.py:181]: I don't think so, the logging capture supports multiple logs being captured at the same time, which is not true for capsys/capfd. :+1:

**Review by nicoddemus — APPROVED** [review]

**Comment by nicoddemus** [comment]
Thanks @rhoban13!

**Pull request merged**